blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/Granularity.java

```diff
@@ -274,9 +288,29 @@ private static Granularity granularityFromPointsGeometric(double requestedDurati
                 break;
             }
         }
+
+        if (gran == null) {
+            gran = Granularity.LAST;
+        }
+
         return gran;
     }
 
+    private static long getTTL(String tenantid, Granularity g) {
+        long ttl;
+        try {
+            ttl = (long)TTL_PROVIDER.getTTL(
```

Comment:
I am confused about which methods we should be using [getTTL](https://github.com/rackerlabs/blueflood/blob/master/blueflood-core/src/main/java/com/rackspacecloud/blueflood/cache/ConfigTtlProvider.java#L125) or [getTTLForIngestion](https://github.com/rackerlabs/blueflood/blob/master/blueflood-core/src/main/java/com/rackspacecloud/blueflood/cache/ConfigTtlProvider.java#L148) and should we consider [areTTLsForced](https://github.com/rackerlabs/blueflood/blob/master/blueflood-core/src/main/java/com/rackspacecloud/blueflood/cache/ConfigTtlProvider.java#L152)